cmd/containerd-shim-runhcs-v1/service_internal.go

```diff
@@ -372,6 +372,60 @@ func (s *service) diagShareInternal(ctx context.Context, req *shimdiag.ShareRequ
 	return &shimdiag.ShareResponse{}, nil
 }
 
+func (s *service) diagTasksInternal(ctx context.Context, req *shimdiag.TasksRequest) (*shimdiag.TasksResponse, error) {
+	raw := s.taskOrPod.Load()
+	if raw == nil {
+		return nil, errors.Wrapf(errdefs.ErrNotFound, "task with id: '%s' not found", s.tid)
+	}
+
+	resp := &shimdiag.TasksResponse{}
+	if s.isSandbox {
+		p, ok := raw.(shimPod)
+		if !ok {
+			return nil, errors.New("failed to convert task to pod")
+		}
+
+		tasks, err := p.GetTasks()
+		if err != nil {
+			return nil, err
+		}
+
+		for _, task := range tasks {
+			t := &shimdiag.Task{ID: task.ID()}
```

Comment:
could be its own function I suppose?

Reply:
+1, i think it can be reused below as well

Reply:
This tasks loop, or the execs portion? The execs portion is the thing shared between two things in this func